### ISSUE TITLE
Fusion gateway can't return list of a union type

### ISSUE THREAD

**Issue by A360JMaxxgamer** [ChilliCream/graphql-platform#6321]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Product

Hot Chocolate

### Describe the bug

I have a subgraph with a type which as a union type list field.

When I query the field via fusion gateway I get an execution error. Switching the union type to an interface type makes the gateway return the values.

It seems, that fusion can not handle union types on list fields.

### Steps to reproduce

1. Create a subgraph with a type which has field with a union type list
2. Query for the type and field on the fusion gateway
3. It should create and error

### Relevant log output

```shell
{
  “errors”: [
    {
      “message”: “Unexpected Execution Error”,
      “extensions”: {
        “message”: “The type structure is invalid.“,
        “stackTrace”: ”   at HotChocolate.Types.TypeExtensions.ListType(IType type)\n   at HotChocolate.Types.TypeExtensions.ElementType(IType type)\n   at HotChocolate.Fusion.Execution.ExecutorUtils.ComposeList(FusionExecutionContext context, ResultData parent, Int32 parentIndex, Selection selection, SelectionData selectionData, IType type)\n   at HotChocolate.Fusion.Execution.ExecutorUtils.ComposeElement(FusionExecutionContext context, ResultData parent, Int32 parentIndex, Selection selection, SelectionData selectionData, IType valueType)\n   at HotChocolate.Fusion.Execution.ExecutorUtils.ComposeList(FusionExecutionContext context, ResultData parent, Int32 parentIndex, Selection selection, SelectionData selectionData, IType type)\n   at HotChocolate.Fusion.Execution.ExecutorUtils.ComposeResult(FusionExecutionContext context, SelectionSet selectionSet, SelectionData[] selectionSetData, ObjectResult selectionSetResult)\n   at HotChocolate.Fusion.Execution.ExecutorUtils.ComposeObject(FusionExecutionContext context, ResultData parent, Int32 parentIndex, ISelection selection, SelectionData selectionData)\n   at HotChocolate.Fusion.Execution.ExecutorUtils.ComposeResult(FusionExecutionContext context, SelectionSet selectionSet, SelectionData[] selectionSetData, ObjectResult selectionSetResult)\n   at HotChocolate.Fusion.Execution.ExecutorUtils.ComposeResult(FusionExecutionContext context, WorkItem workItem)\n   at HotChocolate.Fusion.Planning.Compose.OnExecuteAsync(FusionExecutionContext context, ExecutionState state, CancellationToken cancellationToken)\n   at HotChocolate.Fusion.Planning.QueryPlanNode.ExecuteAsync(FusionExecutionContext context, CancellationToken cancellationToken)\n   at HotChocolate.Fusion.Planning.QueryPlanNode.OnExecuteNodesAsync(FusionExecutionContext context, ExecutionState state, CancellationToken cancellationToken)\n   at HotChocolate.Fusion.Planning.QueryPlanNode.ExecuteAsync(FusionExecutionContext context, CancellationToken cancellationToken)\n   at HotChocolate.Fusion.Planning.QueryPlan.ExecuteAsync(FusionExecutionContext context, CancellationToken cancellationToken)\n   at HotChocolate.Fusion.Execution.FederatedQueryExecutor.ExecuteAsync(FusionExecutionContext context, CancellationToken cancellationToken)\n   at HotChocolate.Fusion.Pipeline.DistributedOperationExecutionMiddleware.InvokeAsync(IRequestContext context, IBatchDispatcher batchDispatcher)\n   at HotChocolate.Execution.Pipeline.OperationVariableCoercionMiddleware.InvokeAsync(IRequestContext context)\n   at HotChocolate.Execution.Pipeline.OperationResolverMiddleware.InvokeAsync(IRequestContext context)\n   at HotChocolate.Execution.Pipeline.OperationComplexityMiddleware.InvokeAsync(IRequestContext context)\n   at HotChocolate.Execution.Pipeline.OperationCacheMiddleware.InvokeAsync(IRequestContext context)\n   at HotChocolate.Execution.Pipeline.DocumentValidationMiddleware.InvokeAsync(IRequestContext context)\n   at HotChocolate.Execution.Pipeline.DocumentParserMiddleware.InvokeAsync(IRequestContext context)\n   at HotChocolate.Execution.Pipeline.DocumentCacheMiddleware.InvokeAsync(IRequestContext context)\n   at HotChocolate.Execution.Pipeline.TimeoutMiddleware.InvokeAsync(IRequestContext context)\n   at HotChocolate.Execution.Pipeline.ExceptionMiddleware.InvokeAsync(IRequestContext context)”
      }
    }
  ]
}
```

### Version

13.3.0

**Comment by michaelstaib** [comment]
Is this still true with 13.4?

**Comment by A360JMaxxgamer** [comment]
Nope. It is fixed with 13.4.  👍
